romancal/multiband_catalog/multiband_catalog_step.py

```diff
@@ -0,0 +1,211 @@
+            "aperture_corrections": np.array((1.0, 1.0, 1.0)),
+            "aperture_ee": aperture_ee,
+            "bkg_aperture_inner_radius": 5.0,
+            "bkg_aperture_outer_radius": 10.0,
+        }
+
+        # TODO: do we want to save the det_img and det_err?
+        det_img, det_err = make_detection_image(library, self.kernel_fwhms)
+
+        # estimate background rms from detection image to calculate a
+        # threshold for source detection
+        mask = np.isnan(det_img)
+        coverage_mask = np.isnan(det_err)
+        bkg = RomanBackground(
+            det_img,
+            box_size=self.bkg_boxsize,
+            mask=mask,
+            coverage_mask=coverage_mask,
+        )
+        bkg_rms = bkg.background_rms
+
+        segment_img = make_segmentation_image(
+            det_img,
+            snr_threshold=self.snr_threshold,
+            npixels=self.npixels,
+            bkg_rms=bkg_rms,
+            deblend=self.deblend,
+            mask=coverage_mask,
+        )
+
+        if segment_img is None:  # no sources found
+            source_catalog_model.source_catalog = Table()
+        else:
+            ci_star_thresholds = (
+                self.ci1_star_threshold,
+                self.ci2_star_threshold,
+            )
+
+            # this is needed for the DAOFind sharpness and roundness
+            # properties; are these needed for the Roman source catalog?
+            star_kernel_fwhm = np.min(self.kernel_fwhms)  # ??
+
+            det_model = maker_utils.mk_datamodel(datamodels.MosaicModel)
+            det_model.data = det_img
+            det_model.err = det_err
+
+            # TODO: this is a temporary solution to get model attributes
+            # currently needed in RomanSourceCatalog
+            with library:
+                model = library.borrow(0)
+                det_model.weight = model.weight
+                det_model.meta = model.meta
+                library.shelve(model, modify=False)
+
+            det_catobj = RomanSourceCatalog(
+                det_model,
+                segment_img,
+                det_img,
+                aperture_params,
+                ci_star_thresholds,
+                star_kernel_fwhm,
+                self.fit_psf,
+                detection_cat=None,
+            )
+            det_cat = update_colnames(det_catobj.catalog, prefix="det_")
+
+            # loop over each image
+            with library:
+                # enumerate is temporary to work around filter name issue
+                for i, model in enumerate(library):
+                    catobj = RomanSourceCatalog(
+                        model,
+                        segment_img,
+                        None,
+                        aperture_params,
+                        ci_star_thresholds,
+                        star_kernel_fwhm,
+                        self.fit_psf,
+                        detection_cat=det_catobj,
+                    )
```

Comment:
Just recording for my thinking---this does totally separate fits on each filter.  e.g., for the PSFs, the centers will jump around a little from filter to filter, or for the krons, they will have separate shapes.

Reply:
No, the Kron shapes are fixed by the Kron radius (with some scaling parameterization), which is calculated only from the detection image.  The initial PSF centers will also be from the detection image centroids (as will the circular aperture centers).

Reply:
We can also do forced PSF photometry with fixed positions (based on the detection image centroids).

Reply:
Thanks, good, I had misunderstood this.  This is good behavior for now.